geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java

```diff
@@ -557,6 +558,42 @@ public boolean virtualPut(EntryEventImpl event, boolean ifNew, boolean ifOld,
     }
   }
 
+  @Override
+  void cmnClearRegion(RegionEventImpl regionEvent, boolean cacheWrite, boolean useRVV) {
+    boolean enableRVV = useRVV && getConcurrencyChecksEnabled();
+    RegionVersionVector rvv = null;
+    if (enableRVV) {
+      rvv = getVersionVector();
+    }
+
+    // lock the primary from moving
+    DistributedLockService lockService = DistributedLockService
+        .getServiceNamed(PartitionedRegionHelper.PARTITION_LOCK_SERVICE_NAME);
+    String lockName = this.getFullPath();
+    while (!lockService.lock(lockName, 100, -1)) {
+      if (!getBucketAdvisor().isPrimary()) {
+        PartitionedRegionException pre =
+            new PartitionedRegionException(
+                "The bucket " + this.getId() + " is no longer a primary. Retry the clear");
+        throw pre;
```

Comment:
Since we locked the primary bucket just now, should we first unlock it before we throw due to the primary no longer being where we expect?